src/callbacks.cpp

```diff
@@ -0,0 +1,231 @@
+                     __func__, e.what());
+        this->job_finished(
+            (std::string) "WaypointNode::" + (std::string) __func__ +
+            ": Received "
+            "invalid json as payload: " +
+            e.what());
+        reset_node();
+        return;
+    }
+
+    // Set flag to true to indicate that the object has values in it
+    cmd.values_set = true;
+
+    // Required parameters
+    cmd.target_coordinate_lat = cmd_json.at("target_coordinate_lat");
+    cmd.target_coordinate_lon = cmd_json.at("target_coordinate_lon");
+    cmd.cruise_height_cm = cmd_json.at("cruise_height_cm");
+    cmd.target_height_cm = cmd_json.at("target_height_cm");
+    cmd.horizontal_speed_mps = cmd_json.at("horizontal_speed_mps");
+    cmd.vertical_speed_mps = cmd_json.at("vertical_speed_mps");
+
+    // Optional parameters
+    if (cmd_json.contains("pre_wait_time_ms"))
+        cmd.pre_wait_time_ms = cmd_json.at("pre_wait_time_ms");
+
+    if (cmd_json.contains("post_wait_time_ms"))
+        cmd.post_wait_time_ms = cmd_json.at("post_wait_time_ms");
+}
+
+/**
+ * @brief Callback function for receiving GPS position messages.
+ *
+ * This function is called when a GPS position message is received. It extracts
+ * the latitude, longitude, and height information from the message and updates
+ * the corresponding variables in the `pos` object.
+ *
+ * @param msg The GPS position message containing the latitude, longitude, and
+ * height information.
+ */
+void WaypointNode::callback_position(const interfaces::msg::GPSPosition &msg) {
+    // Store current timestamp for later
+    rclcpp::Time timestamp_now = this->now();
+
+    RCLCPP_DEBUG(this->get_logger(),
+                 "WaypointNode::%s: Received position from "
+                 "'%s': lat: %f, lon: %f, height: %f",
+                 __func__, msg.sender_id.c_str(), msg.latitude_deg,
+                 msg.longitude_deg, msg.relative_altitude_m);
+
+    // Check timestamp
+    if (timestamp_now - rclcpp::Time(msg.time_stamp) >
+        rclcpp::Duration(std::chrono::duration<int64_t, std::milli>(
+            max_position_msg_time_difference_ms))) {
+        // Message is too old, react based on the current 'active' state
+
+        // Reset pos as we can no longer reliabely know where we are
+        pos = Position();
+
+        if (this->get_active()) {
+            // Abort job if currently active
+
+            RCLCPP_FATAL(get_logger(),
+                         "WaypointNode::%s: Received too old "
+                         "timestamp in position message: %s. Aborting Job.",
+                         __func__, msg.sender_id.c_str());
+            this->job_finished("WaypointNode::" + (std::string) __func__ +
+                               ": A too old timestamp was received in a "
+                               "position message while "
+                               "being active");
+            reset_node();
+        } else {
+            // Warn but still store values
```

Comment:
If the value is too old I would not store it

Reply:
This way I would probably use an even older value that is cached.
So, I rather take an old value instead of an even older one.
Nevertheless, the node is idle anyway in this state so the values are not used anyway.
But as they are needed right on activation, I can not just ignore the values altogether.
I think this is an acceptable compromise.

Reply:
This:
![image](https://github.com/THI-Drone/waypoint_package/assets/62964107/c5341763-3c3c-4efa-9e5f-2cc540b124f0)
**-> Explosion**

Reply:
ok, if thats the plan fine by me